sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/HandlingFiles.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Text;
+using System.IO;
+using Microsoft.Win32.SafeHandles;
+using System.Security.AccessControl;
+using System.IO.Compression;
+using System.IO.IsolatedStorage;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+using System.Security.Policy;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        void Main(IntPtr intPtr, IsolatedStorageFile isolatedStorageFile, MemoryMappedFile memoryMappedFile, MemoryMappedFileSecurity memoryMappedFileSecurity, Stream stream)
+        {
+            // Any static method call on File or Directory will raise
+            File.Exists(""); // Noncompliant {{Make sure this file handling is safe here.}}
+//          ^^^^^^^^^^^^^^^
+            File.Create(""); // Noncompliant
+            File.Delete(""); // Noncompliant
+            File.ReadLines(""); // Noncompliant
+            Directory.Exists(""); // Noncompliant
+            Directory.Delete(""); // Noncompliant
+            Directory.EnumerateFiles(""); // Noncompliant
+
+            // Any FileInfo or DirectoryInfo creation
+            var fileInfo = new FileInfo(""); // Noncompliant {{Make sure this file handling is safe here.}}
+//                         ^^^^^^^^^^^^^^^^
+            var dirInfo = new DirectoryInfo(""); // Noncompliant
+
+            // Calls to extern CreateFile
+            SafeFileHandle handle;
+            handle = CreateFile("", 0, 0, intPtr, 0, 0, intPtr); // Noncompliant
+            handle = CreateFile(); // Compliant, not extern
+
+            // Creation of SafeFileHandle
+            handle = new SafeFileHandle(IntPtr.Zero, false); // Noncompliant
+
+            // All constructors of FileStream
+            FileStream fileStream;
+            fileStream = new FileStream(IntPtr.Zero, FileAccess.Read); // Noncompliant
```

Comment:
There's an exception in the RSPEC for creation of a FileStream from a SafeFileHandle.